### ISSUE TITLE
Unknown transaction input

### ISSUE THREAD

**Issue by luciantugui** [aiken-lang/fortuna#4]
New block not found in 30 seconds, updating state
Found next datum: d8799f1907065820000006dff9205db5e091234e60b34014714a5c9d1fef064b297e3702c04e39420519ffff1a06e8ad401b0000018a3a3957184d416c4c204861496c2074556e419f5820000006dff9205db5e091234e60b34014714a5c9d1fef064b297e3702c04e39425820000000c3eec000141b73d9f9009577b7fe43edd0a5f4eabcbaf88d959e5e562f5820000000c3eec000141b73d9f9009577b7fe43edd0a5f4eabcbaf88d959e5e562f5820000000c3eec000141b73d9f9009577b7fe43edd0a5f4eabcbaf88d959e5e562f58200000006eb7acf12b94c966d0581bbc2801815a3a34194c2e25a8acf85f7bb849582000000022dce0f741c1e9a4429896e2df6c145337dc6b428189de693c4f3e3522582000000002d0da241c8aed392c2afa621fb6b5ac5b4608e239ef7406b68f2ab354582000000002d0da241c8aed392c2afa621fb6b5ac5b4608e239ef7406b68f2ab354582000000002d0da241c8aed392c2afa621fb6b5ac5b4608e239ef7406b68f2ab354582000000002d0da241c8aed392c2afa621fb6b5ac5b4608e239ef7406b68f2ab354ffff
[
  { extraRedeemers: [ "spend:1" ] },
  {
    collectErrors: [
      {
        badTranslation: "Unknown transaction input (missing from UTxO set): 95c06c8d447206a5bdb6ed1042a99bdd6ed5617241abde834"... 17 more characters
      }
    ]
  },
  {
    valueNotConserved: {
      consumed: {
        coins: 10000000,
        assets: {
          "279f842c33eed9054b9e3c70cd6a3b32298259c24b78b895cb41d91a.54554e41": 5000000000
        }
      },
      produced: {
        coins: 12870460,
        assets: {
          "279f842c33eed9054b9e3c70cd6a3b32298259c24b78b895cb41d91a.54554e41": 5000000000,
          "279f842c33eed9054b9e3c70cd6a3b32298259c24b78b895cb41d91a.6c6f72642074756e61": 1
        }
      }
    }
  },
  {
    badInputs: [
      {
        txId: "95c06c8d447206a5bdb6ed1042a99bdd6ed5617241abde8348878d171ab832a3",
        index: 0
      }
    ]
  }
]
Mining...

**Comment by Janilom** [comment]
I have had the same issue for my first hour of mining. Then I mined a block and I received the tokens. I did do a send all to my wallet to make sure there was only 1 UTXO and no collateral. Not sure if that had anything to do with it.

**Comment by jaysnel** [comment]
Having this issue for the last hour or so also, but hasn't resolved itself like @Janilom though

**Comment by Janilom** [comment]
I was able to mint a few blocks but I came back this morning to all the same errors as you have stated. Still an issue for me as well.

**Comment by luciantugui** [comment]
yeah, besides that, sometimes there's also `{ feeTooSmall: { requiredFee: 237426, actualFee: 237206 } }`

**Comment by MicroProofs** [comment]
I'll look into the fee too small. Do you have the tx? The other error (inputs not conserved) comes from other people beating you to mainnet.

**Comment by luciantugui** [comment]
> I'll look into the fee too small. Do you have the tx? The other error (inputs not conserved) comes from other people beating you to mainnet.

[
  { extraRedeemers: [ "spend:0" ] },
  {
    collectErrors: [
      {
        badTranslation: "Unknown transaction input (missing from UTxO set): 07c21cfb9653d42ea673c3a6ccb74f2ad9635442b101a2a16"... 17 more characters
      }
    ]
  },
  {
    valueNotConserved: {
      consumed: {
        coins: 3449351,
        assets: {
          "279f842c33eed9054b9e3c70cd6a3b32298259c24b78b895cb41d91a.54554e41": 330000000000
        }
      },
      produced: {
        coins: 6319811,
        assets: {
          "279f842c33eed9054b9e3c70cd6a3b32298259c24b78b895cb41d91a.54554e41": 330000000000,
          "279f842c33eed9054b9e3c70cd6a3b32298259c24b78b895cb41d91a.6c6f72642074756e61": 1
        }
      }
    }
  },
  {
    badInputs: [
      {
        txId: "07c21cfb9653d42ea673c3a6ccb74f2ad9635442b101a2a16c42dee274861f54",
        index: 0
      }
    ]
  },
  {
    totalCollateralMismatch: { computedFromDelta: 1104711, declaredInField: 748902 }
  },
  { feeTooSmall: { requiredFee: 237426, actualFee: 237206 } }
]

**Comment by MicroProofs** [comment]
> > I'll look into the fee too small. Do you have the tx? The other error (inputs not conserved) comes from other people beating you to mainnet.
> 
> 
> 
> how or why others beat me? resources?

Yes you are competing to be the the first user to output a new datum and spend the old one. If you lose the race your tx fails.

**Comment by MicroProofs** [comment]
> Getting the same, maybe being front-run? would also love how to see how others are beating me too
> 
> ```
> 
> [
> 
>   { extraRedeemers: [ "spend:1" ] },
> 
>   {
> 
>     collectErrors: [
> 
>       {
> 
>         badTranslation: "Unknown transaction input (missing from UTxO set): d956cd84c86b6d52a2368d0bd9d418806e48eefa55b0f0968"... 17 more characters
> 
>       }
> 
>     ]
> 
>   },
> 
>   {
> 
>     valueNotConserved: {
> 
>       consumed: {
> 
>         coins: 6902181,
> 
>         assets: {
> 
>           "279f842c33eed9054b9e3c70cd6a3b32298259c24b78b895cb41d91a.54554e41": 5000000000
> 
>         }
> 
>       },
> 
>       produced: {
> 
>         coins: 9772641,
> 
>         assets: {
> 
>           "279f842c33eed9054b9e3c70cd6a3b32298259c24b78b895cb41d91a.54554e41": 5000000000,
> 
>           "279f842c33eed9054b9e3c70cd6a3b32298259c24b78b895cb41d91a.6c6f72642074756e61": 1
> 
>         }
> 
>       }
> 
>     }
> 
>   },
> 
>   {
> 
>     badInputs: [
> 
>       {
> 
>         txId: "d956cd84c86b6d52a2368d0bd9d418806e48eefa55b0f09685500b591bf2361c",
> 
>         index: 0
> 
>       }
> 
>     ]
> 
>   }
> 
> ]
> 
> ```

Same issue here you lost the race. Other miners in the future will have a more graceful error message.

**Comment by luciantugui** [comment]
> I'll look into the fee too small. Do you have the tx? The other error (inputs not conserved) comes from other people beating you to mainnet.

I constantly get that (my guess it's something to do with the wallet)

[
  { extraRedeemers: [ "spend:0" ] },
  {
    collectErrors: [
      {
        badTranslation: "Unknown transaction input (missing from UTxO set): 006b172d48cb4dfe638e0324f8192a2121a5c03df7ffcbce7"... 17 more characters
      }
    ]
  },
  {
    valueNotConserved: {
      consumed: {
        coins: 3449351,
        assets: {
          "279f842c33eed9054b9e3c70cd6a3b32298259c24b78b895cb41d91a.54554e41": 330000000000
        }
      },
      produced: {
        coins: 6263781,
        assets: {
          "279f842c33eed9054b9e3c70cd6a3b32298259c24b78b895cb41d91a.54554e41": 330000000000,
          "279f842c33eed9054b9e3c70cd6a3b32298259c24b78b895cb41d91a.6c6f72642074756e61": 1
        }
      }
    }
  },
  {
    badInputs: [
      {
        txId: "006b172d48cb4dfe638e0324f8192a2121a5c03df7ffcbce70b7ae6038f2fd0d",
        index: 0
      }
    ]
  },
  {
    totalCollateralMismatch: { computedFromDelta: 1104711, declaredInField: 748902 }
  },
  { feeTooSmall: { requiredFee: 237426, actualFee: 237206 } }
]

**Comment by Janilom** [comment]
How can we win more ? Local instances of cardano node?

**Comment by luciantugui** [comment]
> I'll look into the fee too small. Do you have the tx? The other error (inputs not conserved) comes from other people beating you to mainnet.

another one

Found next datum: d8799f190c5f5820000003e0651d3a80a95a66380e9581da0163ffec8ffa613cf630b74bf8116aae05193fff1a02bf6e201b0000018a3d6aa1a84d416c4c204861496c2074556e419f582000000180dadf4816aba9c81f2e1d3c3852b6fe81f0ce38e17f228f2e461cedf65820000000253cd8f92d6bd1b9aae1350d1d58d7f8c8b6875025868a4d2eaf515c455820000000253cd8f92d6bd1b9aae1350d1d58d7f8c8b6875025868a4d2eaf515c455820000000253cd8f92d6bd1b9aae1350d1d58d7f8c8b6875025868a4d2eaf515c4558200000001e12f2b3520828b59ca594bb38c66c8706f9cb84b2f85ea58fc8d985fd58200000000ce5e15adb830826704e141ba76269de7a852b5b6a706f5d3e51c9b393582000000000a00e4dc59365fe4bfa11bce8b41110c56f698e046c06f148ba4f2eb0582000000000a00e4dc59365fe4bfa11bce8b41110c56f698e046c06f148ba4f2eb0582000000000a00e4dc59365fe4bfa11bce8b41110c56f698e046c06f148ba4f2eb0582000000000a00e4dc59365fe4bfa11bce8b41110c56f698e046c06f148ba4f2eb0ffff
[
  {
    totalCollateralMismatch: { computedFromDelta: 1104711, declaredInField: 745951 }
  },
  { feeTooSmall: { requiredFee: 239393, actualFee: 239173 } }
]

**Comment by MicroProofs** [comment]
hmm add a console.log to print out the tx and I can take a look at that

**Comment by MicroProofs** [comment]
You can set native_uplc to false in the lucid complete() function to possibly fix this right now.

**Comment by luciantugui** [comment]
`AlL HaIl tUnA`

a small PR would be nice 💯

**Comment by luciantugui** [comment]
usually this happens when ada on THAT address is too low, I am transferring all ada from wallet to that address and it's fixed

**Comment by jabteles** [comment]
> usually this happens when ada on THAT address is too low, I am transferring all ada from wallet to that address and it's fixed

What do you consider "not low"? Is 40A enough? I'm getting this error even with 40A with no Collateral set.

Will not transfer big amounts of ADA to an experimental project which created the wallet and probably knows its seed :p
